etl/steps/data/garden/ihme_gbd/2023-10-04/leading_causes_child_deaths.py

```diff
@@ -0,0 +1,214 @@
+from typing import Any, List
+
+import owid.catalog.processing as pr
+from owid.catalog import Table
+from owid.catalog.utils import underscore
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+# naming conventions
+paths = PathFinder(__file__)
+log = get_logger()
+
+
+def run(dest_dir: str) -> None:
+    # Load in the cause of death data and hierarchy of causes data
+    tb_cause = paths.load_dataset("gbd_cause")
+    tb_hierarchy = paths.load_dataset("cause_hierarchy")
+    tb_hierarchy = tb_hierarchy["cause_hierarchy"]
+
+    # Underschore the hierachy cause names to match tb_cause
+    tb_hierarchy["cause_name_underscore"] = tb_hierarchy["cause_name"].apply(underscore)
+    tb_hierarchy = add_owid_hierarchy(tb_hierarchy)
+    # We'll iterate through each level of the hierachy to find the leading cause of death in under-fives in each country-year
+    levels = [1, 2, 3, 4, "owid_all_ages", "owid_under_5"]
+    age_groups = ["under_5", "all_ages"]
+
+    tb_out = []
+    for level in levels:
+        log.info(f"Processing level {level}")
+        for age_group in age_groups:
+            log.info(f"Processing age group {age_group}")
+            # Get the causes at this level
+            level_causes = tb_hierarchy[tb_hierarchy["level"] == level]["cause_name_underscore"].to_list()
+            # Create table with leading cause of death at this level for each country-year
+            tb_level = create_hierarchy_table(
+                age_group=age_group,
+                tb_cause=tb_cause,
+                level_causes=level_causes,
+                short_name=f"leading_cause_level_{level}_in_{age_group}",
+            )
```

Comment:
Because of constructing the table this way, `tb_level` doesn't have any origins. The origins are added on the next step, by `clean_disease_names`. This means that in the end you only preserve the origin 'Global Burden of Disease Study - Cause Hierarchy'. If that's the expected result, then all good.